### DISCUSSION TITLE
使用createForm嵌套渲染子表单

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#1320] [Q&A 2.x] [answered]
在一个自定义组件中使用createForm嵌套渲染子表单，子表单下的field路径会被带上自定义组件本身的路径作为基路径。

子表单作为一个独立的Form实例，路径应该不受父级影响。

![image](https://user-images.githubusercontent.com/18549980/116353527-85644a80-a829-11eb-8c35-6ce819730eff.png)

codesandbox ：

https://codesandbox.io/s/mutipleform-m50xp?file=/App.tsx

**Comment by NexxLuo** [comment]
通过把SchemaField的basePath设置为[""]解决了这个问题，另外发现一个问题`basePath:""` 这样设置无效；
`   <SchemaField schema={current} basePath={[""]}></SchemaField>`

**Comment by janryWang** [comment, marked as answer]
这个问题是因为FormProvider没有对Context做清理，下个版本修复该问题
